buildSrc/build.gradle

```diff
@@ -107,6 +107,7 @@ dependencies {
   compile localGroovy()
   
   compile 'commons-codec:commons-codec:1.12'
+  compile 'org.apache.commons:commons-compress:1.19'
```

Comment:
Perhaps I'm missing something but I don't seem to see any usages of this new dependencies as part of this PR.

Reply:
@mark-vieira See https://github.com/elastic/elasticsearch/pull/46470/files#diff-47425746d987cda11e39635561e213ddR234-R297. I think that you didn't expand the diff?

Reply:
Also, I have discovered there is more work to do here, this preserves the symbolic links when exploding the tar (which `tarTree` did not do), but `Tar` tasks do not preserve symbolic links either (from the source), so I am working on pushing a change to this PR to write our own `Tar` task to preserve those. 🤦‍♀

Reply:
What's the behavior we get instead? Does gradle simply replace the symlinks with the actual fine when creating a tar including symlinks? Is this problematic when creating our tar distribution with a bundled JDK? Is it worth the complexity of maintaining our own `Tar` implementation?

Reply:
When exploding a tar archive, Gradle completely drops the symlink. When executing a copy that includes symlinks, Gradle copies the target instead of preserving the symlink (exploding the size of the resulting destination). If you override this to preserve the symlinks (by adding the code I added to the JDK copy spec), the `Tar` task drops the symlinks on the floor. These are all known issues that have lived for years.

Reply:
Got it, so it's definitely something about the `Tar` implementation specifically that doesn't deal with symlinks. We should try and pressure them to sort this out given that AdoptOpenJDK is becoming increasingly popular and itself uses symlinks. Thanks for the explanation.